### PR TITLE
[Full-Page-Cache] Fix: Wrong links in multi site system

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/6199

### CODE REVIEW

**Comment by blankse** [comment]
I add isFrontend/isSiteRequest to the Snippet cache keys. This params are the indicators in the getPath method to switch between backend/frontend links.

I also remove the try/catch. This should be only part of the __toString() method. With the render() method it should be handled by symfony. See https://github.com/pimcore/pimcore/pull/6042 and https://github.com/pimcore/pimcore/pull/4996

File: models/Document/Tag/Snippet.php [hunk 1]
```diff
@@ -114,71 +115,61 @@ public function frontend()
...
+            if ($this->snippet->getUseTargetGroup()) {
+                $cacheParams['target_group'] = $this->snippet->getUseTargetGroup();
             }
 
-            return $content;
-        } catch (\Exception $e) {
-            Logger::error($e);
+            $cacheParams['webp'] = Frontend::hasWebpSupport();
+            $cacheParams['frontend'] = Tool::isFrontend();
+            $cacheParams['siteRequest'] = Site::isSiteRequest();
```
Comment (brusch) [inline models/Document/Tag/Snippet.php:151]: I think we should better use the ID of the site here, then `isFrontend()` shouldn't be necessary.

Comment (blankse) [reply models/Document/Tag/Snippet.php]: I changed it

**Comment by blankse** [comment]
@brusch Are there still any doubts?

**Comment by brusch** [comment]
@blankse Sorry, haven't got the time yet for a detailed review and testing.

**Comment by brusch** [comment]
Looks good, thanks a lot!

**Pull request merged**